srtcore/buffer_rcv.cpp

```diff
@@ -254,53 +254,52 @@ int CRcvBuffer::dropAll()
     return dropUpTo(end_seqno);
 }
 
-int CRcvBuffer::dropMessage(int32_t seqnolo, int32_t seqnohi, int32_t msgno)
+int CRcvBuffer::dropMessage(int32_t seqnolo, int32_t seqnohi, int32_t msgno, DropActionIfExists actionOnExisting)
 {
     IF_RCVBUF_DEBUG(ScopedLog scoped_log);
     IF_RCVBUF_DEBUG(scoped_log.ss << "CRcvBuffer::dropMessage: seqnolo " << seqnolo << " seqnohi " << seqnohi << " m_iStartSeqNo " << m_iStartSeqNo);
+    const bool bKeepExisting = (actionOnExisting == KEEP_EXISTING);
     // TODO: count bytes as removed?
-    const int end_pos = incPos(m_iStartPos, m_iMaxPosOff);
-    if (msgno > 0) // including SRT_MSGNO_NONE and SRT_MSGNO_CONTROL
+    int minDroppedOffset = -1;
+    int iDropCnt = 0;
+    if (msgno > 0) // excluding SRT_MSGNO_NONE and SRT_MSGNO_CONTROL
     {
+        // First try to drop by message number in case the message starts earlier thtan @a seqnolo.
         IF_RCVBUF_DEBUG(scoped_log.ss << " msgno " << msgno);
-        int minDroppedOffset = -1;
-        int iDropCnt = 0;
+        const int end_pos = incPos(m_iStartPos, m_iMaxPosOff);
```

Comment:
If `m_iStartPos` isn't going to be modified inside this branch, then moving this declaration here was not necessary, as it's still necessary in the further part of the code and is defined identically.